hledger/test/journal/valuation.test

```diff
@@ -360,9 +360,9 @@ $ hledger -f- bal -N --value=2000-01-15 -b 2000
 $ hledger -f- bal -N --value=now -b 2000
                 12 B  a
 
-# 33. single column balance report valued at default date (same as --value=end)
-$ hledger -f- bal -N -V -b 2000
-                12 B  a
+# 33. single column balance report valued at cost and end (should be the same as --cost in this case)
+$ hledger -f- bal -N --cost --value=end -b 2000
+                21 B  a
 
```

Comment:
What drove this change ? Is it related to the rest of the PR ? Perhaps just a minor fixup, but I'm struggling to grok things.

Reply:
This was because I initially had an implementation which passed all the tests, but on closer inspection would have failed this test. I noticed we didn't have any tests where both cost and valuation are done, so I added one. The old test here was really a duplicate of a previous one (same code accessed by different flags), so I thought replacing it was a better plan than renumbering all the tests.

Reply:
Moved into a separate PR.